### DISCUSSION TITLE
Changing Modules directory

### DISCUSSION THREAD

**Discussion by caglarkarahan** [openssl/openssl#24483] [Q&A] [answered]
I am working on building openssl 3.3.0 with legacy providers which can be enabled with enable-weak-ssl-ciphers. It creats legacy.so library in MODULESDIR. But I want to change this directory according to looking path where I moved the libraries. Is it possible to cahnge modules directory? If so, how can I do it? It needs to look different path in my case.

**Comment by mattcaswell** [comment, marked as answer]
MODULESDIR can be changed either at OpenSSL build time, or at runtime.

At OpenSSL build time the MODULESDIR is of the form *prefix*/*libdir*/ossl-modules. Where "prefix" defaults to "/usr/local", and libdir depends on the build but might typically be "lib64". You can change the prefix using "--prefix=*whatever*" in your config/Configure command line. This changes the prefix for all the OpenSSL installation (i.e. not just MODULESDIR).

At runtime you can set the MODULESDIR environment variable to override the build time default.

**Reply by lili2012** [reply]
for me it is OPENSSL_MODULES environment variable
